release/ray_release/tests/test_config.py

```diff
@@ -39,6 +41,47 @@
     }
 )
 
+VALID_TEST_DEFINITION = TestDefinition(
+    **{
+        "name": "validation_test",
+        "working_dir": "validation_dir",
+        "frequency": "nightly",
+        "team": "release",
+        "cluster": {
+            "cluster_env": "app_config.yaml",
+            "cluster_compute": "tpl_cpu_small.yaml",
+        },
+        "run": {
+            "timeout": 100,
+            "script": "python validate.py",
+        },
+        "flavors": [
+            {
+                "name": "aws",
+            },
+            {
+                "name": "gce",
+                "cluster": {
+                    "cluster_env": "app_config.yaml",
+                    "cluster_compute": "tpl_cpu_small_gce.yaml",
+                },
+            },
+        ],
+    }
+)
+
+
+def test_definition_parser():
+    test_definition = VALID_TEST_DEFINITION.copy()
+    tests = parse_test_definition([test_definition])
+    aws_test = tests[0]
+    gce_test = tests[1]
+    schema = load_schema_file()
+    assert not validate_test(aws_test, schema)
+    assert not validate_test(gce_test, schema)
+    assert aws_test["name"] == "validation_test-aws"
```

Comment:
is it good/desired to have a mix of underscore and dash?

Reply:
Discuss in another commend, I'm open to use any other type of delimiter, but I think some form of consistent delimiter is good